File: Models/NrwcongTy.cs
using System;
using System.Collections.Generic;

namespace WebAPI_NRW.Models;

public partial class NrwcongTy
{
    public int Id { get; set; }

    public string Ma { get; set; } = null!;

    public int Ky { get; set; }

    public int Nam { get; set; }

    public double SanLuongDauVao { get; set; }

    public double SanLuongTieuThu { get; set; }

    public double? LuongNuocThatThoat { get; set; }

    public double? TyLeThatThoatChuan2 { get; set; }

    public double? TyLeThatThoatChuan1 { get; set; }

    public DateOnly TuNgay { get; set; }

    public DateOnly DenNgay { get; set; }

    public int? SoNgayDocSoDht { get; set; }

    public double? SoNgayDocSoBilling { get; set; }

    public string? NguyenNhan { get; set; }

    public string? GhiChu { get; set; }

    public DateTime NgayTao { get; set; }

    public DateTime? NgayCapNhat { get; set; }

    public string? NguoiTao { get; set; }

    public string? NguoiCapNhat { get; set; }

    public virtual NguoiDung? NguoiCapNhatNavigation { get; set; }

    public virtual NguoiDung? NguoiTaoNavigation { get; set; }

    public virtual ICollection<NrwcongTyDauVaoChiTiet> NrwcongTyDauVaoChiTiets { get; set; } = new List<NrwcongTyDauVaoChiTiet>();

    public virtual ICollection<NrwcongTyTieuThuChiTiet> NrwcongTyTieuThuChiTiets { get; set; } = new List<NrwcongTyTieuThuChiTiet>();
}

File: src/types/nrw-cong-ty/nrw-cong-ty.ts
/* ========== REQUEST (CLIENT --> API) ========== */
// Lấy tổng sản lượng
export interface TongSanLuongRequest {
  TuNgay: string;
  DenNgay: string;
  MaDoiTuong: number;
}

// Lấy sản lượng tiêu thụ
export interface SanLuongTieuThuRequest {
  Ky: number;
  Nam: number;
  MaDoiTuong: number;
}

// Lấy sản lượng tiêu thụ
export interface SoNgayDocSoBillingRequest {
  Ky: number;
  Nam: number;
}

// Thêm dữ liệu
export interface AddNrwCongTyRequest {
  Ma: string;
  Ky: number;
  Nam: number;
  SanLuongDauVao: number;
  SanLuongTieuThu: number;
  LuongNuocThatThoat?: number;
  TyLeThatThoatChuan2?: number;
  TyLeThatThoatChuan1?: number;
  TuNgay: string;
  DenNgay: string;
  SoNgayDocSoDht?: number;
  SoNgayDocSoBilling?: number;
  NguyenNhan?: string;
  GhiChu?: string;
  NgayTao: string;
  NguoiTao: number;
}

// Cập nhật dữ liệu
export interface UpdateNrwCongTyRequest {
  Ma: string;
  Ky: number;
  Nam: number;
  SanLuongDauVao: number;
  SanLuongTieuThu: number;
  LuongNuocThatThoat?: number;
  TyLeThatThoatChuan2?: number;
  TyLeThatThoatChuan1?: number;
  TuNgay: string;
  DenNgay: string;
  SoNgayDocSoDht?: number;
  SoNgayDocSoBilling?: number;
  NguyenNhan?: string;
  GhiChu?: string;
  NgayCapNhat?: string;
  NguoiCapNhat?: number;
}

/* ========== RESPONE (API --> CLIENT) ========== */
// Dữ liệu trả về
export interface NrwCongTyResponse {
  id: number;
  ma: string;
  ky: number;
  nam: number;
  sanLuongDauVao: number;
  sanLuongTieuThu: number;
  luongNuocThatThoat?: number;
  tyLeThatThoatChuan2?: number;
  tyLeThatThoatChuan1?: number;
  tuNgay: string;
  denNgay: string;
  soNgayDocSoDht?: number;
  soNgayDocSoBilling?: number;
  nguyenNhan?: string;
  ghiChu?: string;
  ngayTao: string;
  ngayCapNhat?: string;
  nguoiTao?: string;
  nguoiCapNhat?: string;
}

// Lấy tổng sản lượng
export interface TongSanLuongRespone {
  tongSanLuong: number;
}

// Lấy sản lượng tiêu thụ
export interface SanLuongTieuThuRespone {
  sanLuong: number;
}

// Lấy số ngày đọc số Billing
export interface SoNgayDocSoBillingRespone {
  soNgayDocSoBilling: number;
}